### DISCUSSION TITLE
Dagster+ Launcher role permissions for a subset of assets and jobs

### DISCUSSION THREAD

**Discussion by cody-scott** [dagster-io/dagster#24623] [Q&A] [answered]
Question regarding dagster+ permissions for the launcher roles. 

It doesn't look possible based on the docs, but what would be the best approach to allow a subset of jobs/assets to be launched by a subset of users? 

It seems like spinning them out to their own code location would be the cleanest approach. I see a downside in that we are using `.downstream()` to call the entry asset then all downstream assets afterwards (such as dbt). Splitting across code locations would break this too i think.

Given that is the code location the best (current) approach to allow a group of users the ability to launch a limited selection of jobs?

https://docs.dagster.io/dagster-plus/account/managing-users/managing-user-roles-permissions#general

**Comment by garethbrickman** [comment, marked as answer]
Hey Cody, at this time it's not possible to set a Launcher role at the granular level of assets or jobs so you would have to manage that with multiple code locations. We are in the process of developing improvements to the customizability of roles, which could lay the groundwork for more granular level customization in future.

**Reply by cody-scott** [reply]
Sounds like what I suspected! Thanks @garethbrickman
